src/common/stringops.cpp

```diff
@@ -27,6 +27,51 @@
 // implementation
 // ===========================================================================
 
+#if wxUSE_UNICODE_WCHAR && SIZEOF_WCHAR_T == 2
+
+wxStringOperationsWchar::Utf16CharBuffer wxStringOperationsWchar::EncodeChar(const wxUniChar& ch)
+{
+    wxUint32 v = ch.GetValue();
+    Utf16CharBuffer buf = {{0}};
+    if ( v < 0x10000 )
+    {
+        buf.data[0] = (wchar_t)v;
+    }
+    else
+    {
+        buf.data[0] = (wchar_t)(0xD800 | ((v - 0x10000) >> 10));
```

Comment:
This code is not rocket science, of course, but it still duplicates (in a slightly different form) the existing `{en,de}code_utf16()` functions from `src/common/strconv.cpp`.

Would it be possible to reuse them, e.g. just extract them into inline functions in some private header (existing or new `wx/private/utf16.h`, as you prefer)?